dev/documentation/en/getting-started.md

```diff
@@ -103,19 +103,19 @@ Run the following commands whenever you want to pull the latest changes to your
 ```
 # Update the app
 git checkout main
-git pull upstream main
+git pull --rebase upstream main
 git push
 
 # Update the pkp-lib submodule
 cd lib/pkp
 git checkout main
-git pull upstream main
+git pull --rebase upstream main
 git push
 
 # Update the ui-library submodule
 cd ../ui-library
 git checkout main
-git pull upstream main
+git pull --rebase upstream main
```

Comment:
We don't want to recommend using `--rebase` here or the three commands above. Using `rebase` will replay local commits on the top of the `main` branch, but this will cause conflicts down the road.

Instead, when developing with the main application (when expecting to commit back), developers are encouraged to _never modify the `main` branch_. All changes need to be done in a separate branch.

If a `git checkout main && git pull upstream main` can not run without a rebase, your local copy is out of sync with PKP's copy and will lead to problems if/when your local changes are merged (or not).

Reply:
About your 2nd paragraph, what you state is already part of the **[contributions section](https://docs.pkp.sfu.ca/dev/documentation/en/getting-started#contributions)**. This is just the installation section.

 When using `git pull --rebase`, only the hash of your local commits changes and not the remote ones. If rebase raises conflicts, the developer can solve them and use git merge instead. Local rebase avoids those unnecessary merge commits in the history like 'Merge branch X in Y'.

Reply:
The only merge commit messages should be `Merge pull request #XXX from YYY` from the upstream repo, since these ones have the PR number link  with the reviews and comments.